lib/msal-common/src/server/CodeRequestParameters.ts

```diff
@@ -0,0 +1,429 @@
+            }
+        }
+    }
+
+    /**
+     * Check to see if there are SSO params set in the Request
+     * @param request
+     */
+    isSSOParam(account: Account) {
+        const isSSORequest = this.userRequest && (this.userRequest.account || this.userRequest.sid || this.userRequest.loginHint);
+        return account || isSSORequest;
+    }
+
+    /**
+     * @hidden
+     * @ignore
+     * 
+     * Utility to populate QueryParameters and ExtraQueryParameters to ServerRequestParamerers
+     * @param adalIdTokenObject 
+     */
+    populateQueryParams(adalIdTokenObject?: any): void {
+        let queryParameters: StringDict = {};
+
+        if (this.userRequest) {
+            // add the prompt parameter to serverRequestParameters if passed
+            if (this.userRequest.prompt) {
+                this.validatePromptParameter(this.userRequest.prompt);
+            }
+
+            // Add claims challenge to serverRequestParameters if passed
+            if (this.userRequest.claimsRequest) {
+                validateClaimsRequest(this.userRequest);
+            }
+
+            // if the developer provides one of these, give preference to developer choice
+            if (this.isSSOParam(this.userRequest.account)) {
+                queryParameters = this.constructUnifiedCacheQueryParameter(null);
+            }
+        }
+
+        if (adalIdTokenObject) {
+            queryParameters = this.constructUnifiedCacheQueryParameter(adalIdTokenObject);
+        }
+
+        /*
+         * adds sid/login_hint if not populated; populates domain_req, login_req and domain_hint
+         * this.logger.verbose("Calling addHint parameters");
+         */
+        queryParameters = this.addHintParameters(queryParameters);
+
+        // sanity check for developer passed extraQueryParameters
+        let eQParams: StringDict;
+        if (this.userRequest) {
+            eQParams = this.sanitizeEQParams(this.userRequest);
+        }
+
+        // Populate the extraQueryParameters to be sent to the server
+        this.queryParameters = this.generateQueryParametersString(queryParameters);
+        this.extraQueryParameters = this.generateQueryParametersString(eQParams);
+    }
+
+    /**
+     * Create navigation url.
+     */
+    async createNavigateUrl(): Promise<string> {
+        const str = await this.createParamString();
+        let authEndpoint: string = this.authorityInstance.authorizationEndpoint;
+        // if the endpoint already has queryparams, lets add to it, otherwise add the first one
+        if (authEndpoint.indexOf("?") < 0) {
+            authEndpoint += "?";
+        } else {
+            authEndpoint += "&";
+        }
+
+        const requestUrl: string = `${authEndpoint}${str.join("&")}`;
+        return requestUrl;
+    }
+
+    /**
+     * Create a query parameter string.
+     */
+    private async createParamString(): Promise<Array<string>> {
```

Comment:
would it be better to use template strings here instead of a bunch of `str.push()`? or do we need these different values in an array somewhere else, like for validity checks?

Reply:
I think this design was used for code readability, but if you have any suggestions for how to improve it I am open to them. These parameters will have been validated by the time this function is called.

Reply:
i'm not sure if others think it's better or more readable, but you'd end up concatenating template literals together to save readability. as an example: 
```
const templateLiteral = `abcdefgh` +
    `ijklmnopqrst` +
    `uvwxyz`;
```

it's really not too different, just a thought.

Reply:
I'm not sure if the readability improves with that, and I don't think we are losing anything by using the string array to build the params. interested to hear other's thoughts.

Reply:
We'll call it fine how it is, no reason to make more work. Resolving